packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -26,38 +26,55 @@ const getFunctionMetadataMap = async (): Promise<
 > => {
   const filepaths = glob
     .sync(path.join(FUNCTIONS_PATH, "**/*.{js,ts}"), { nodir: true })
-    .map((f) => path.resolve(f));
-  const functionMetadataArray: [string, FunctionMetadata][] = await Promise.all(
-    filepaths.map(async (filepath) => {
-      const buildResult = await esbuild.build({
-        entryPoints: [filepath],
-        outdir: TEMP_DIR,
-        write: false,
-        format: "esm",
-        bundle: true,
-      });
-      const importedFile = await importFromString(
-        buildResult.outputFiles[0].text
-      );
-      const relativePath = path.relative(process.cwd(), filepath);
-      return [relativePath, { entrypoint: importedFile.default?.name }];
-    })
-  );
+    .map(f => path.resolve(f))
+  const functionMetadataArray: [string, FunctionMetadata][] = []
+
+  await Promise.allSettled(filepaths.map(generateFunctionMetadata)).then(
```

Comment:
instead of `.then` I think using await and  then a `.filter` like `const results = await Promise.allSettled()` `const blah = results.filter(d => can console.error here)` is a little cleaner